src/main/java/org/dsa/iot/knx/DeviceFolder.java

```diff
@@ -146,29 +132,48 @@ public Node buildFolderTree(Node parent, Queue<String> path) {
 		}
 	}
 
-	public void buildDataPoint(Node parent, GroupAddress address, String dataSize, String mainGroupName,
-			String middleGroupName, String subGroupName) {
+	@Override
+	public void buildDataPoint(Node parent, GroupAddressBean addressBean) {
+		GroupAddress address = null;
+		try {
+			address = new GroupAddress(addressBean.getGroupAddress());
+		} catch (KNXFormatException e) {
+			e.printStackTrace();
+		}
 		int mainGroupAddress = address.getMainGroup();
 		int middleGroupAddress = address.getMiddleGroup();
-		int subGroupAddress = address.getSubGroup11();
+		GroupAddressType groupLevel = getConnection().getGroupLevel();
+		int subGroupAddress = 0;
+		if (groupLevel == GroupAddressType.THREE_LEVEL) {
+			subGroupAddress = address.getSubGroup8();
+		} else if (groupLevel == GroupAddressType.TWO_LEVEL) {
+			subGroupAddress = address.getSubGroup11();
+		}
 
-		PointType type = getDataTypeByDataSize(dataSize);
-		String name = mainGroupName + "_" + middleGroupName + "_" + subGroupName;
+		PointType type = getDataTypeByDataSize(addressBean.getDataSize());
 		ValueType valueType = getValueType(type);
 
-		Node pointNode = parent.createChild(name).setValueType(valueType).build();
+		Node pointNode = parent.createChild(addressBean.getName()).setValueType(valueType).build();
 		pointNode.setAttribute(ATTR_POINT_TYPE, new Value(type.toString()));
-		pointNode.setAttribute(ATTR_MAIN_GROUP_NAME, new Value(mainGroupName));
-		pointNode.setAttribute(ATTR_MIDDLE_GROUP_NAME, new Value(middleGroupName));
-		pointNode.setAttribute(ATTR_SUB_GROUP_NAME, new Value(subGroupName));
 		pointNode.setAttribute(ATTR_MAIN_GROUP_ADDRESS, new Value(mainGroupAddress));
 		pointNode.setAttribute(ATTR_MIDDLE_GROUP_ADDRESS, new Value(middleGroupAddress));
 		pointNode.setAttribute(ATTR_SUB_GROUP_ADDRESS, new Value(subGroupAddress));
 		pointNode.setAttribute(ATTR_RESTORE_TYPE, new Value(ATTR_EDITBALE_POINT));
 
-		KnxConnection conn = getConnection();
 		DevicePoint knxPoint = new DevicePoint(conn, this, pointNode);
-		conn.getGroupToPoint().put(middleGroupName, knxPoint);
+		updateGroupToPoints(addressBean.getMiddleGroup(), knxPoint);
+	}
+
+	private void updateGroupToPoints(String group, DevicePoint point) {
+		ArrayList<EditablePoint> points = null;
+		Map<String, ArrayList<EditablePoint>> groupToPoints = getConnection().getGroupToPoints();
```

Comment:
I feel like this is at the wrong place. Looks like a violation of tell don't ask to me.